pymongo/auth.py

```diff
@@ -545,5 +545,6 @@ def authenticate(credentials, sock_info):
     """Authenticate sock_info."""
     mechanism = credentials.mechanism
     auth_func = _AUTH_MAP.get(mechanism)
-    auth_func(credentials, sock_info)
+    if auth_func:
+        auth_func(credentials, sock_info)
```

Comment:
I think the idea here is that mechanism should always be one of the keys in _AUTH_MAP. Let's use `_AUTH_MAP[mechanism]` instead of `.get()` to make that more clear.

Reply:
Done